agent_plugin_builder/build_options.py

```diff
@@ -35,7 +38,25 @@ class AgentPluginBuildOptions(InfectionMonkeyBaseModel):
     ]
 
 
-def parse_agent_plugin_build_options(file: Path) -> AgentPluginBuildOptions:
-    with open(file, "r") as f:
-        data = yaml.safe_load(f)
-        return AgentPluginBuildOptions(**data)
+def parse_agent_plugin_build_options(plugin_path: Path) -> AgentPluginBuildOptions:
+    """
+    Parse the build options for an agent plugin from the plugin's directory.
+
+    :param plugin_path: The path to the plugin code directory.
+    """
+    build_config_file_path = plugin_path / "build.yaml"
+    if not build_config_file_path.exists():
+        build_config_file_path = plugin_path / "build.yml"
+
+    if not build_config_file_path.exists():
+        logger.info(
+            "Build options not found, using default dependency method: "
+            f"{PlatformDependencyPackagingMethod.SEPARATE}"
+        )
+        return AgentPluginBuildOptions(
+            platform_dependencies=PlatformDependencyPackagingMethod.SEPARATE
+        )
```

Comment:
NIT: To avoid unnecessary revisions if we ever add new options, we could be more vague here and just say "Using default options". We can also then just do:

```suggestion
    if not build_config_file_path.exists():
        logger.info(
            "Build options not found, using default dependency method: "
            f"{PlatformDependencyPackagingMethod.SEPARATE}"
        )
        return AgentPluginBuildOptions()
```